jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/RefreshAwsTree.kt

```diff
@@ -5,15 +5,26 @@ package software.aws.toolkits.jetbrains.core.explorer
 
 import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.project.Project
+import software.aws.toolkits.jetbrains.core.AwsResourceCache
 import software.aws.toolkits.jetbrains.core.Resource
 import software.aws.toolkits.jetbrains.core.clearResourceForCurrentConnection
+import software.aws.toolkits.jetbrains.core.credentials.ConnectionSettings
 
-fun Project.refreshAwsTree(resource: Resource<*>? = null) {
-    if (resource == null) {
-        this.clearResourceForCurrentConnection()
+fun Project.refreshAwsTree(resource: Resource<*>? = null, connectionSettings: ConnectionSettings? = null) {
+    if (connectionSettings == null) {
+        if (resource == null) {
+            this.clearResourceForCurrentConnection()
+        } else {
+            this.clearResourceForCurrentConnection(resource)
+        }
     } else {
-        this.clearResourceForCurrentConnection(resource)
+        if (resource == null) {
+            AwsResourceCache.getInstance().clear(connectionSettings)
+        } else {
+            AwsResourceCache.getInstance().clear(resource, connectionSettings)
+        }
     }
+
```

Comment:
I am not sure this is is the best route for this since in this branch it will still refresh the tree even though it shouldn't be.

We should probably have a higher level util that takes care of do this if connectionSetts still match, else only invalidate cache

Reply:
what if an else if condition is added instead of else? Also, do you mean we don't refresh the AWS explorer just invalidate the cache?